service/common/src/main/java/org/apache/polaris/service/catalog/generic/GenericTableCatalogHandler.java

```diff
@@ -94,7 +99,14 @@ public LoadGenericTableResponse createGenericTable(
             createdEntity.getFormat(),
             createdEntity.getDoc(),
             createdEntity.getPropertiesAsMap());
-
+    if (XTableConversionUtils.requiresConversion(callContext, properties)) {
+      XTableConverter converter =
+          new XTableConverter(XTableConversionUtils.getHostUrl(callContext));
```

Comment:
Can we register a singleton in the framework instead of creating an instance per call?

Reply:
make sense